[DRAFT - experimental] Migrate from next12 pages to next13 app dir

annaharr opened the pull request:
- Moved pages out of pages directory and into new next13 app directory
  - pages/index.js is now app/page.js
  - Head and app container from _app.js now is now in app/head.js and app/layout.js respectively 
  - Note that all pages in app dir are server rendered unless "use client" is specified 
  
- Created a DataProvider
  - DataProvider now fetches data at 6 second intervals from s3 bucket 
  - DataContextProvider wraps all app pages in app/layout.js
  - app/page.js and app/data/page.js get sample data from DataContext

- Global styles moved from pages/global.css to styles/globals.css

Commit:
app dir migration

Commit:
feat: working on adding a disclaimer to the data page

Commit:
create data provider

Commit:
root page uses data context

Commit:
migrate to app folder structure

Commit:
feat: Disclaimer

- Added a Disclaimer to the Data top bar
- Updated the Data top bar styles to account for the disclaimer
- Created disclaimer styles to be responsive

Commit:
add container for css

Commit:
nivo version change

Commit:
data polling from provider

Commit:
clean up and add useCallback

Commit:
Move Navbar to components

Commit:
removed defaultProps

Commit:
Merge branch 'disclaimer/link' into next-13-app-dir

Commit:
update endpoint to use constants

jedahan reviewed (approved):
This looks great - i just put some small cleanup suggestions in the other PR to this PR

Commit:
hydration error fix

Commit:
hydration fix

Commit:
App dir restructuring

Commit:
navbar link update

Commit:
update data template and container classnames

Commit:
update content nav dashboard link

Commit:
add navbar to loader

Commit:
updated to usePathName with new next13 app directory

Commit:
refactor: deleted useRouter from navbar

Commit:
css cleanup

Commit:
page state to boolean showBanner

Commit:
regenerate package.lock

Commit:
fetch data interval changed to 10 min

AKaudinov committed:
refactor: DateRangePicker window size hydration issue fix (#147)

- Created a new hook: useIsMobile
   - which executes on the client side
- Refactored DateRangePicker to a functional component so
  it can use the new `useIsMobile` hook when mounting
- Removed the usage of isMobile layout helper that was trying
  to access the window width directly on React hydration after
  server render, which caused different rendered content issues.
   - the new isMobile hook uses `useEffect` which fires after
     mount, preventing hydration mismatches

Co-authored-by: Amir Kaudinov <amirkaudinov@R3060.local>

AKaudinov commented:
Keeping this PR open as an experimental draft of Next.js 13 experimental app directory
